### PR TITLE
{Misc} Remove some Python 2.7 legacy code

### PR DESCRIPTION
**Pull request by atombrella**
**Description**

Removed a bit of Python 2.7 code that I found by searching for `ImportError` and `Python 2.7`.

There are still quite a fair pieces of Python 2.7 compatibility around. `sys.version_info`  is a good place to start.

Note that I also saw a fair chunk of https://docs.python.org/3/library/datetime.html#datetime.datetime.utcnow in the code

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [X] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [X] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [X] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
Misc

File: src/azure-cli/azure/cli/command_modules/keyvault/vendored_sdks/azure_keyvault_securitydomain/_serialization.py [hunk 1]
```diff
@@ -53,12 +53,13 @@
...
 )
 
-try:
-    from urllib import quote  # type: ignore
-except ImportError:
-    from urllib.parse import quote
+from urllib.parse import quote
 import xml.etree.ElementTree as ET
 
+from datetime import timezone as _FixedOffset  # type: ignore
```
Comment (atombrella) [inline src/azure-cli/azure/cli/command_modules/keyvault/vendored_sdks/azure_keyvault_securitydomain/_serialization.py:59]: I'll fix this. The SDK has this serialization file duplicated in quite a few directories.

**Comment by atombrella** [comment]
@wangzelin007 @jiasli I won't add more to this PR. `_serialization.py` is weird.

To summarize better what this PR contains:

I fixed some of the warnings from the pyupgrade section from ruff.

https://docs.astral.sh/ruff/rules/open-alias/
https://docs.astral.sh/ruff/rules/os-error-alias/
https://docs.astral.sh/ruff/rules/native-literals/
https://docs.astral.sh/ruff/rules/unnecessary-future-import/

Apart from that, I have grepped all over the code to remove ImportError for obvious Python 2 libraries; mostly that was `urllib`.

This one can also be fixed, but I didn't do so. There are two matches:
https://docs.astral.sh/ruff/rules/replace-stdout-stderr/

This one finds a few matches, but pylint doesn't complain since it was fixed. Maybe because those few instances are a bit special.
https://docs.astral.sh/ruff/rules/unicode-kind-prefix/

**Comment by jiasli** [comment]
@atombrella, first, thanks for the great work. Do you mind creating separate PRs for each change, such as the `IOError` deprecation, so that we can easily review your PR? Thanks again!

**Comment by atombrella** [comment]
> @atombrella, first, thanks for the great work. Do you mind creating separate PRs for each change, such as the `IOError` deprecation, so that we can easily review your PR? Thanks again!

@jiasli :) I've opened some PRs that should be more granular in terms of changes.

I'll close this PR.

**Pull request closed without merging**